pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -169,6 +172,69 @@ cpdef int64_t delta_to_nanoseconds(delta) except? -1:
     raise TypeError(type(delta))
 
 
+cdef str npy_unit_to_abbrev(NPY_DATETIMEUNIT unit):
+    if unit == NPY_DATETIMEUNIT.NPY_FR_ns or unit == NPY_DATETIMEUNIT.NPY_FR_GENERIC:
+        # generic -> default to nanoseconds
+        return "ns"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_us:
+        return "us"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_ms:
+        return "ms"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_s:
+        return "s"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_m:
+        return "m"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_h:
+        return "h"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_D:
+        return "D"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_W:
+        return "W"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_M:
+        return "M"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_Y:
+        return "Y"
+    else:
+        raise NotImplementedError(unit)
+
+
+cdef object ensure_td64ns(object ts):
```

Comment:
isn't this duplicating: https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/tslibs/conversion.pyx#L253

Reply:
that is for ndarray, not td64 scalar.  i havent found a good way to share that